src/components/Facility/PatientNotesList.tsx

```diff
@@ -53,18 +52,20 @@ const PatientNotesList = (props: PatientNotesProps) => {
         totalPages: Math.ceil(data.count / pageSize),
       }));
     }
-    setIsLoading(false);
-    setReload(false);
   };
 
   useEffect(() => {
     if (reload) {
-      fetchNotes();
+      fetchNotes().then(() => {
+        setIsLoading(false);
+        setReload(false);
+      });
```

Comment:
This looks hacky to me, consider having setIsLoading and setReload in fetchNotes itself, wrap fetchNotes in useCallback, and handle error with try catch if needed.

cc: @rithviknishad can u comment on this

Reply:
I agree with @khavinshankar. 

As an alternative I'd recommend sticking with what `useQuery` provides for managing the loading state. Consider building a `useInfiniteQuery` hook if necessary to manage merging queries across from multiple page responses.